test/functional/ui/screen.lua

```diff
@@ -187,22 +187,29 @@ end
 --              "{K:S}", where K is a key in `attr_ids`. Any unexpected
 --              attributes in the final state are an error.
 -- attr_ignore: Ignored text attributes, or `true` to ignore all.
--- condition:   Function asserting some arbitrary condition.
+-- condition:   Function asserting some arbitrary condition. Can also be
+--              supplied as only argument.
 -- any:         true: Succeed if `expected` matches ANY screen line(s).
 --              false (default): `expected` must match screen exactly.
 function Screen:expect(expected, attr_ids, attr_ignore, condition, any)
   -- remove the last line and dedent
-  expected = dedent(expected:gsub('\n[ ]+$', ''))
   local expected_rows = {}
-  for row in expected:gmatch('[^\n]+') do
-    -- the last character should be the screen delimiter
-    row = row:sub(1, #row - 1)
-    table.insert(expected_rows, row)
-  end
-  if not any then
-    assert(self._height == #expected_rows,
-      "Expected screen state's row count(" .. #expected_rows
-      .. ') differs from configured height(' .. self._height .. ') of Screen.')
+  if type(expected) ~= "string" then
+    condition = expected
+    expected = nil
```

Comment:
Should also throw an error if more than one arg was passed, to have some level of sanity checking.